widget/button.go

```diff
@@ -167,7 +166,7 @@ func (b *Button) MouseOut() {
 
 // SetIcon updates the icon on a label - pass nil to hide an icon
 func (b *Button) SetIcon(icon fyne.Resource) {
-	b.Icon = icon
+	b.Icon = cache.OverrideResourceTheme(icon, b)
```

Comment:
Hmm, I feel like having a widget public `XYZ` field and the `SetXYZ` function result in `wid.XYZ` being something other than what was passed into `SetXYZ` is confusing behavior. I don't think we have a precedent for `wid.SetXYZ(f00)` being in any way different, behavior-wise, from `wid.XYZ = foo; wid.Refresh()`?

Reply:
This is simply a shortcut required to adopt a resource when being set on a button. It replaces the prior "ForWidget" field on the resource.
If setting an icon resource manually and the developer wants it to reflect theme override it could be called manually as it's public api.

I'm not sure what the alternative is. If you have suggestions please say. As far as I can see we either link the resource to the widget, or we wrap the resource when the widget adopts it. I could certainly remove this line of code for a "clean setter" but then developers would have to manually adopt the resource for the widget. Is that better?

Reply:
This also isn't something custom widgets will be able to do. That can be problematic.

Reply:
I'm not following what custom widgets can't do?

Reply:
I guess the alternative could be calling `cache.OverrideResourceTheme` on the icon in the Refresh function? There could be other alternatives, I just think that we should stick to the invariant that `wid.SetFoo(bar)` results in `wid.Foo == bar` always (where there is both a publicly exported `Foo` property and the setter)

Reply:
Won't that infinitely wrap the resource?
Either way it will result in the field access not returning the same as Set was passed. Potentially worse the field will change over time...

Reply:
I was referring to having a new private field `themedIcon` or similar in the button, that on Refresh, we set `btn.themedIcon = cache.OverrideResourceTheme(btn.Icon, btn)`. Then the button uses `themedIcon` internally instead of `Icon` directly

Reply:
Ah thank you. Let me explore that.